generate-icons.js

```
@@ -0,0 +1,158 @@
+    await generateIcon({
+      size,
+      padding: 0,
+      inputFilename: SETTINGS.iconLightFilename,
+      outputFilename: `favicon-dark-mode-${size}x${size}.png`,
+      background: transparent,
+    });
+  });
+
+  const faviconBuffer = await pngToIco(SETTINGS.faviconSizes.map((size) => path.join(__dirname, SETTINGS.outputDirectory, `favicon-${size}x${size}.png`)));
+  await fs.writeFile(path.join(__dirname, SETTINGS.outputDirectory, 'favicon.ico'), faviconBuffer);
+
+  const faviconDarkBuffer = await pngToIco(SETTINGS.faviconSizes.map((size) => path.join(__dirname, SETTINGS.outputDirectory, `favicon-dark-mode-${size}x${size}.png`)));
+  await fs.writeFile(path.join(__dirname, SETTINGS.outputDirectory, 'favicon-dark-mode.ico'), faviconDarkBuffer);
+
+// MS Tile icons - these are more faffy as regards padding and sizes
+  await generateIcon({
+    size: 144,
+    padding: 0,
+    inputFilename: SETTINGS.iconLightFilename,
+    outputFilename: 'mstile-144x144.png',
+    background: transparent,
+  });
+
+  await generateIcon({
+    width: 128,
+    height: 81,
+    padding: 16,
+    inputFilename: SETTINGS.iconLightFilename,
+    outputFilename: 'mstile-70x70.png',
+    background: transparent,
+  });
+
+  await generateIcon({
+    width: 270,
+    height: 208,
+    xPadding: 70,
+    yPadding: 50,
+    inputFilename: SETTINGS.iconLightFilename,
+    outputFilename: 'mstile-150x150.png',
+    background: transparent,
+  });
+
+  await generateIcon({
+    width: 558,
+    height: 208,
+    xPadding: 216,
+    yPadding: 50,
+    inputFilename: SETTINGS.iconLightFilename,
+    outputFilename: 'mstile-310x150.png',
+    background: transparent,
+  });
+
+  await generateIcon({
+    width: 558,
+    height: 430,
+    xPadding: 150,
+    yPadding: 128,
+    inputFilename: SETTINGS.iconLightFilename,
+    outputFilename: 'mstile-310x310.png',
+    background: transparent,
+  });
+} catch (error) {
+  console.error('Error generating icons:', error);
```

Shouldn't catch this error - let it fail